### PR TITLE
Docs/update migrate state instructions

### PR DESCRIPTION
**Pull request by trujillo-adam**
### What
This PR updates the instructions for migrating existing TF state files to the cloud platforms. The original request identified a missing field in the configuration and that the content was oriented toward HCP TF only. 
As part of the update, I refactored the information so that it aligns better with our usage content type format. I also provided minimal instructions for using the API to complete the action. In the future, we should provide more complete instructions, including example request calls and request body configurations. Although it is inconsistent with our documentation guidelines, I left the link to the external example for now.   

[Preview](https://terraform-docs-common-git-docs-update-migrate-62747d-hashicorp.vercel.app/terraform/cloud-docs/migrate)

#### Pull Request
- [X] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. HCP Terraform ).
- [X] Description links to related pull requests or issues, if any.

#### Content
- [X] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/migrate/index.mdx [hunk 1]
```diff
@@ -1,43 +1,79 @@
...
-```
-terraform {
-  cloud {
-    organization = "my-org"
-    workspaces {
-      tags = ["networking"]
+1. Add the `cloud` block to your Terraform configuration and specify the following fields:
+   1. `hostname` field: Specify either `app.terraform.io` for HCP Terraform or the hostname of your Terraform Enterprise deployment. 
+   1. `organization` field: Specify your HCP Terraform or Terraform Enterprise organization. 
+   1. `workspaces` block: Add the `tags` field and specify one or more destination workspaces as a list of strings.
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/migrate/index.mdx:37]: Should we mention that the `workspaces` block can defined the `tags` field OR the `name` field? I did some digging around and it looks like `tags` is the recommended way, according to this CLI output:

```
│ Only one of workspace "tags" or "name" is allowed.
│
│ The 'workspaces' block configures how Terraform CLI maps its workspaces for this single
│ configuration to workspaces within an HCP Terraform or Terraform Enterprise organization. Two
│ strategies are available:
│
│ tags - A set of tags used to select remote HCP Terraform or Terraform Enterprise
│ workspaces to be used for this single
│ configuration. New workspaces will automatically be tagged with these tag values. Generally, this
│ is the primary and recommended strategy to use.  This option conflicts with "name".
│
│ name - The name of a single HCP Terraform or Terraform Enterprise workspace to be
│ used with this configuration.
│ When configured, only the specified workspace can be used. This option conflicts with "tags"
│ and with the TF_WORKSPACE environment variable.
```

Might just be how I read it, but this makes me feel like I can _only_ use the `tags` field when migrating my state.

File: website/docs/cloud-docs/migrate/index.mdx [hunk 2]
```diff
@@ -1,43 +1,79 @@
...
 
--> **Note:** The `cloud` block is available in Terraform v1.1 and later. Previous versions can use the [`remote` backend](/terraform/language/settings/backends/remote) to configure the CLI workflow and migrate state.
+## Migrate state using the API
 
-## API Migration
+1. Encode your state files as a base64 string and generate an MD5 hash. The following example generates the string and hash file for a single `terraform.tfstate` file:
 
-You can use the HCP Terraform API to automate migrating many state files at once.
+   ```shell-session
+    $ echo terraform.tfstate | base64
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/migrate/index.mdx:61]:
```suggestion
    $ cat terraform.tfstate | base64
```

File: website/docs/cloud-docs/migrate/index.mdx [hunk 3]
```diff
@@ -1,43 +1,79 @@
...
-  cloud {
-    organization = "my-org"
-    workspaces {
-      tags = ["networking"]
+1. Add the `cloud` block to your Terraform configuration and specify the following fields:
+   1. `hostname` field: Specify either `app.terraform.io` for HCP Terraform or the hostname of your Terraform Enterprise deployment. 
+   1. `organization` field: Specify your HCP Terraform or Terraform Enterprise organization. 
+   1. `workspaces` block: Add the `tags` field and specify one or more destination workspaces as a list of strings. 
+
+   The following example below shows how you can map your CLI workspaces to HCP Terraform workspaces that have a specific tag.
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/migrate/index.mdx:39]:
```suggestion
   Refer to [The `cloud` Block](/terraform/cli/cloud/settings#the-cloud-block) in the Terraform CLI documentation for additional information. The following example migrates the state associated with the configuration to the `networking` workspace on HCP Terraform:
```

File: website/docs/cloud-docs/migrate/index.mdx [hunk 4]
```diff
@@ -1,43 +1,79 @@
...
+   - [Create a state version](/terraform/cloud-docs/api-docs/state-versions#create-a-state-version) in the HCP Terraform API reference documentation. 
+   - [Create a state version](/terraform/enterprise/api-docs/state-versions#create-a-state-version) in the Terraform Enterprise API reference documentation.
+1. Send a `POST` request to the `/workspaces/:workspace_id/actions/unlock` endpoint to unlock the workspace. 
 
-This blog post contains an example of how to automate state migration with the API: [Migrating A Lot of State with Python and the HCP Terraform (previously Terraform Cloud) API](https://medium.com/hashicorp-engineering/migrating-a-lot-of-state-with-python-and-the-terraform-cloud-api-997ec798cd11). The example uses the [Workspaces API](/terraform/cloud-docs/api-docs/workspaces#create-a-workspace) to create the necessary workspaces in HCP Terraform and the [State Versions API](/terraform/cloud-docs/api-docs/state-versions) to migrate the state files to those workspaces.
+Refer to the following external article for an example how to create a script in Python to automate multiple state files:
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/migrate/index.mdx:78]:
```suggestion
Refer to the following external article for an example of how to create a script in Python to automate multiple state files:
```

**Pull request merged**